Rubberduck.Core/UI/Command/Refactorings/CodePaneRefactorMoveToFolderCommand.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Refactorings.MoveToFolder;
+using Rubberduck.UI.Command.Refactorings.Notifiers;
+using Rubberduck.VBEditor.Utility;
+
+namespace Rubberduck.UI.Command.Refactorings
+{
+    public class CodePaneRefactorMoveToFolderCommand : RefactorCodePaneCommandBase
+    {
+        private readonly RubberduckParserState _state;
+        private readonly ISelectedDeclarationProvider _selectedDeclarationProvider;
+
+        public CodePaneRefactorMoveToFolderCommand(
+            MoveToFolderRefactoring refactoring,
+            MoveToFolderRefactoringFailedNotifier failureNotifier,
+            ISelectionProvider selectionProvider,
+            RubberduckParserState state,
+            ISelectedDeclarationProvider selectedDeclarationProvider)
+            : base(refactoring, failureNotifier, selectionProvider, state)
+        {
+            _selectedDeclarationProvider = selectedDeclarationProvider;
+            _state = state;
+
+            AddToCanExecuteEvaluation(SpecializedEvaluateCanExecute);
+        }
+
+        private bool SpecializedEvaluateCanExecute(object parameter)
+        {
+            var target = GetTarget();
+
+            return target is ModuleDeclaration 
+                   && !_state.IsNewOrModified(target.QualifiedModuleName);
```

Comment:
Would expect this logic to exactly match the corresponding method in `CodePaneRefactorMoveContainingFolderCommand`.

Reply:
Does this refer to the `null` check? Otherwise, I do not understand this comment. Could you elaborate?

Reply:
It does - the code is otherwise identical :)